### ISSUE TITLE
@Default does not work with @Permission

### ISSUE THREAD

**Issue by carelesshippo** [JorelAli/CommandAPI#197]
**CommandAPI version**
5.10

**Minecraft version**
1.16.4

**Describe the bug**
When using the annotations, having different permissions for default commands does not work.

**My code**
This code reproduces the behavior:
```java
@Command("s")
@Alias("smps")
public class SpectatorCommand {
    @Default
    @Permission("smpspectator.use")
    public static void s(Player player) {
        plugin.getSpectatorManager().togglePlayer(player);
    }

    @Default
    @Permission("smpspectator.force")
    public static void sForce(CommandSender sender, @APlayerArgument Player target) {
        plugin.getSpectatorManager().togglePlayer(target, true);
    }
}
```

Decompiled:
```java
(new CommandAPICommand("s"))
      .withAliases(new String[] { "smps" }).executesPlayer((sender, args) -> SpectatorCommand.s(sender))
      
      .register();
    (new CommandAPICommand("s"))
      .withAliases(new String[] { "smps" }).withArguments(new Argument[] { (Argument)new PlayerArgument("target") }).executes((sender, args) -> SpectatorCommand.sForce(sender, (Player)args[0]))
```

**Expected behavior**
Permissions would be added to the default commands based on the arguments

**Comment by JorelAli** [comment]
I'll investigate this issue within the next 48 hours. If it happens to be a relatively simple fix, I'll try and include it in the next update (due to release very soon)

**Comment by JorelAli** [comment]
TL;DR: The CommandAPI doesn't let you handle this case because arguments can't have permissions.

-----

## Why permissions don't work with @Default

So say we have your class "spectator". This has the following command syntax:

```
/s
/s <target>
/smps
/smps <target>
```

And you've implemented this as following (

```java
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import dev.jorel.commandapi.annotations.Alias;
import dev.jorel.commandapi.annotations.Command;
import dev.jorel.commandapi.annotations.Default;
import dev.jorel.commandapi.annotations.Permission;
import dev.jorel.commandapi.annotations.arguments.APlayerArgument;

@Command("s")
@Alias("smps")
public class SpectatorCommand {
	
	@Default
	@Permission("smpspectator.use")
	public static void s(Player player) {
		// Code here
	}

	@Default
	@Permission("smpspectator.force")
	public static void sForce(CommandSender sender, @APlayerArgument Player target) {
		// Code here
	}
	
}
```

Compiling this (if you're using maven, the output is present in `./target/generated-sources/annotations/SpectatorCommand$Command.java`) produces this class:

```java
import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.PlayerArgument;

import org.bukkit.entity.Player;

// This class was automatically generated by the CommandAPI
public class SpectatorCommand$Command {

    @SuppressWarnings("unchecked")
    public static void register() {

        new CommandAPICommand("s")
            .withAliases("smps")
            .executesPlayer((sender, args) -> {
                // Code here
            })
            .register();

        new CommandAPICommand("s")
            .withAliases("smps")
            .withArguments(new PlayerArgument("target"))
            .executes((sender, args) -> {
                // Code here
            })
            .register();

        }

}
```

You're trying to add two different permissions to the same command - in this case, your command is the `/s` command. Bukkit doesn't allow you to have two permissions for the same command. However, the CommandAPI _can_ handle this case by applying the permission _to the argument_:

```java
new CommandAPICommand("s")
    .withAliases("smps")
    .withPermission("smpspectator.use")
    .executesPlayer((sender, args) -> {
        // Code here
    })
    .register();

new CommandAPICommand("s")
    .withAliases("smps")
    .withPermission("smpspectator.use")
    .withArguments(new PlayerArgument("target").withPermission("smpspectator.force"))
    .executes((sender, args) -> {
        // Code here
    })
    .register();

}
```

The CommandAPI's annotation system doesn't let you add permissions to arguments yet. This is planned to be implemented at some point in the whole "redo the annotation system to make it as powerful as the non-annotation system". For more information about when this is likely to happen, feel free to peruse the [project timeline](https://github.com/JorelAli/CommandAPI#future-project-plans--timeline) at your leisure. At the time of writing, this is scheduled for v5.14.

-----

I have stated this briefly in the documentation that the annotation system is less feature-rich than registering commands normally and that the annotation framework is in its infancy, so the short-term solution is simply to use the normal registration method (without annotations)

**Comment by carelesshippo** [comment]
Thanks for your help!
